### PR TITLE
Use snyk on travis-ci

### LINKED ISSUE #3742: Use Snyk in place of nsp to test for known security vulnerabilities
### Describe the problem and steps to reproduce it:

When a known security vulnerability in one of our dependencies is disclosed, we want our tests to fail so we can apply the patch or ignore the advisory.

### What happened?

We already use [`nsp`](https://github.com/nodesecurity/nsp) for this but it doesn't catch all advisories.

### What did you expect to happen?

We should catch as many advisories as possible. Let's try switching to [`Snyk`](https://snyk.io/) because it reports more advisories.

### LINKED ISSUE #5750: The Node Security Platform service is shutting down 9/30
See: https://blog.npmjs.org/post/175511531085/the-node-security-platform-service-is-shutting

I think we should remove `nsp` from the list of dependencies and also remove the job run on Travis CI. We have Snyk for security checks, which seems more accurate and does not randomly fail.

What do you think?

### PR DESCRIPTION
**Pull request by willdurand**
Fix #5750
Fix #3742

---

After having merged this PR, I am not sure we still need the Pull Request integration.
I used the recommended way to setup Snyk in a CI context: https://snyk.io/docs/ci

### CODE REVIEW

**Comment by willdurand** [comment]
> I see this in the docs but not in your patch. Do we not need it?

I have added the environment variable on Travis CI.

> Also, what is the process for applying fixes from Synk? Will it submit a PR to us? I have found their patches to be very invasive. That is, they change unecessary things in package.json like moving dev deps into prod deps. We'll have to be careful when merging any of their PRs.

I don't think it will do anything for us, thus avoiding their invasive/weird PRs.

After having run `snyk auth` once, we can run `yarn snyk` locally and see how to fix the problem (still locally). Then, we can submit a PR.

File: package.json [hunk 1]
```diff
@@ -27,6 +26,7 @@
     "prettier": "prettier --write '**'",
     "prettier-ci": "npm run prettier -- --list-different || (echo '\n\nThis failure means you did not run `yarn prettier-dev` before committing\n\n' && exit 1)",
     "prettier-dev": "pretty-quick --branch master",
+    "snyk": "snyk test --severity-threshold=medium --org=add-ons-team --file=package.json && snyk monitor --org=add-ons-team",
```
[the remark this answers is not in the document]
Comment (willdurand) [reply package.json]: This command requires authentication, so I did not want to expose it to our contributors.

**Comment by kumar303** [comment]
> [When `yarn snyk` fails in CI ...] After having run snyk auth once, we can run yarn snyk locally and see how to fix the problem (still locally). Then, we can submit a PR.

I'd like to see this process documented so that people other than you know how to do it.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (willdurand) [reply package.json]: 😅

File: README.md [hunk 2]
```diff
@@ -44,10 +44,12 @@ Here are some commands you can run:
 | yarn flow:check             | Explicitly check for Flow errors and exit                                                                  |
 | yarn flow:dev               | Continuously check for Flow errors                                                                         |
 | yarn eslint                 | Lint the JS                                                                                                |
+| yarn snyk                   | Run [snyk][] (without a command)                                                                           |
```
Comment (kumar303) [inline README.md]: You could also just link these to the Synk section with [anchors](https://gist.github.com/asabaylus/3071099).

File: README.md [hunk 3]
```diff
@@ -165,6 +167,20 @@ export default ConfirmButton;
...
+
+We use [snyk][] to continuously monitor our application's dependencies.
+
+As a member of the `add-ons-team`, you can fix an issue reported by running:
+
+```
+yarn snyk-wizard
+```
+
+The wizard allows you to decide whether you want to upgrade dependencies or ignore the issue for 30 days. See the existing reasons to ignore an issue in the [`.snyk`](.snyk) file. Snyk is a bit intrusive and changes many things (like re-adding `snyk test` to the npm `test` script): double check your changes before submitting a Pull Request. You have successfully fixed an issue when `yarn snyk-ci` does not complain.
```
Comment (kumar303) [inline README.md]: Oh, I get it now! I like this workflow.

**Pull request merged**